### ISSUE TITLE
System should know the rules for Extended Warranties and apply them on the fly (Smart Choice Protection is sold based on Sales Volume)

### ISSUE THREAD

**Issue by kentbower** [kbower/tickettest1#673]
_Imported from trac ticket 673.
Created by **rneville**
Opened in trac: Feb 10, 2011, 3:01:00 PM
Last modified in trac: Mar 18, 2011, 2:39:09 PM_
***
I take it these are special SKUs? Are there different SKUs? What are the rules for applying them?

**Comment by kentbower** [comment]
_**rneville** trac comment on Feb 11, 2011, 9:38:49 AM:_

```
Mariam: The SKUS are 'special' but they're not defined as TREATMENT skus. They're NLT, 
non-locatable + taxable item. The way we sell our treatment is based on merchandise amount.
 
If you buy UP TO $ 700 the sku used should be SMART0700
If you buy UP TO $ 1500  the sku used should be SMART1500
If you buy UP TO $ 2000  the sku used should be SMART2000
If you buy UP TO $ 2500  the sku used should be SMART2500
If you buy UP TO $ 3750  the sku used should be SMART3750
If you buy UP TO $ 5000  the sku used should be SMART5000
If you buy UP TO $ 10000  the sku used should be SMART10000
If you buy UP TO $ 20000  the sku used should be SMART20000
 
The most IMPORTANT part here, is that we do not want to allow our sales associates 
to pick which sku they will use. The system should give them the treatment SKU that 
is required to cover the dollar amount in merchandise they have on that reservation.
```
OK, this shouldn't be a problem. I assume this is applied to the entire order? When determining the "up to" portion, are they things I should not be counting in that (for example tax, delivery charges, setup charges, etc...)

**Comment by kentbower** [comment]
_**rneville** trac comment on Feb 23, 2011, 10:04:30 AM:_

Mariam: You are correct in your assumption that this is applied to the entire order; this includes any backorder derived from the original.
The up to portion is merchandise (inventory type sku) only. Tax, delivery & any other charges are not  included in this calculation.  I like your suggestion that no matter which one they try to add, you always set it back to the correct sku.

**Comment by kentbower** [comment]
_**rneville** trac comment on Feb 25, 2011, 2:22:32 PM:_

adding Mariam as CC

**Comment by kentbower** [comment]
_**rneville** trac comment on Mar 8, 2011, 2:53:29 PM:_

Mariam, for calculating the $ amount to compare against the different tiers of warranty prices, am I to total up INV items on the order before or after any promotions and/or discounts are applied?

**Comment by kentbower** [comment]
_**rneville** trac comment on Mar 8, 2011, 3:33:53 PM:_

Does the discount(s) (I'm thinking of especially dollar discounts applied to
the whole order total) then not apply to the warranty? Otherwise if you're
close to a threshold between tiers of warranties you're kind of in a
circular reference problem...which warranty to use can only be determined
after figuring out how the discounts fall but you can only do that after
adding the warranty - which you don't know which one to use.

**Comment by kentbower** [comment]
_**mariam** trac comment on Mar 8, 2011, 4:26:19 PM:_

Q: Does the discount(s) (I'm thinking of especially dollar discounts applied to the whole order total) then not apply to the warranty?

A: The discount is never applied to the warranty. We have to figure out the total merchandise dollars, including discounts, before we figure out which warranty to apply.

For instance, if you have an order that totals $1000 in merchandise (not including warranty) and qualifies for $301 off, the customer will have to pay the price of a warranty that covers up to $699.
The price for that warranty is $99 and it will never get a discount nor will it count towards the $1000 required to get the $301 off.

Did I answer your question? No worries if I didn't, just let me know.

Thanks!

Mariam H. Farach
Project Manager

El Dorado Furniture
corp. offices: 4200 NW 167th St., Miami Gardens, FL, 33054
direct: 305.624.9700 ext. 55411
e-mail: mariam@eldoradofurniture.com
Visit us online! www.eldoradofurniture.comhttp://www.eldoradofurniture.com/

From: Rob Neville [mailto:rn@retailarchitects.com]
Sent: Tuesday, March 08, 2011 3:33 PM
To: Mariam Farach
Cc: issues
Subject: Re: Ticket #673, Status=accepted: System should know the rules for Extended Warranties and apply them on the fly (Smart Choice Protection is sold based on Sales Volume)

Does the discount(s) (I'm thinking of especially dollar discounts applied to the whole order total) then not apply to the warranty? Otherwise if you're close to a threshold between tiers of warranties you're kind of in a circular reference problem...which warranty to use can only be determined after figuring out how the discounts fall but you can only do that after adding the warranty - which you don't know which one to use.
On Tue, Mar 8, 2011 at 3:28 PM, Mariam Farach <Mariam@eldoradofurniture.commailto:Mariam@eldoradofurniture.com> wrote:
Hi!
Sorry, I hadn't seen that email. The total should be AFTER any discounts or promotions...

Thanks!

Mariam H. Farach
Project Manager

El Dorado Furniture
corp. offices: 4200 NW 167th St., Miami Gardens, FL, 33054
direct: 305.624.9700 ext. 55411
e-mail: mariam@eldoradofurniture.commailto:mariam@eldoradofurniture.com
Visit us online! www.eldoradofurniture.comhttp://www.eldoradofurniture.com/

From: Rob Neville [mailto:rn@retailarchitects.com<mailto:rn@retailarchitects.com>]
Sent: Tuesday, March 08, 2011 3:09 PM
To: Mariam Farach
Subject: Fwd: Ticket #673, Status=accepted: System should know the rules for Extended Warranties and apply them on the fly (Smart Choice Protection is sold based on Sales Volume)

Just wanted to make sure you saw this question...
---------- Forwarded message ----------
From: Product Development@RetailArchitects <issues@retailarchitects.commailto:issues@retailarchitects.com>
Date: Tue, Mar 8, 2011 at 2:53 PM
Subject: Ticket #673, Status=accepted: System should know the rules for Extended Warranties and apply them on the fly (Smart Choice Protection is sold based on Sales Volume)
To:
Ticket #673 (accepted defect)
System should know the rules for Extended Warranties and apply them on the fly (Smart Choice Protection is sold based on Sales Volume)
Owner:

rneville

Reporter:

rneville

Milestone:

None

Priority:

blocker

Severity:

None

Comments: (by rneville)
Mariam, for calculating the $ amount to compare against the different tiers of warranty prices, am I to total up INV items on the order before or after any promotions and/or discounts are applied?

________________________________

Ticket URL: https://trac.retailarchitects.com:444/trac/ticket/673
trac https://trac.retailarchitects.com:444/trac
RMS



--
Rob Neville
RetailArchitects
(888) 891-4511 x801
-----------------------------------------------
For Support: (888) 891-4511 Option 2 or support@retailarchitects.commailto:support@retailarchitects.com



--
Rob Neville
RetailArchitects
(888) 891-4511 x801
-----------------------------------------------
For Support: (888) 891-4511 Option 2 or support@retailarchitects.commailto:support@retailarchitects.com

**Comment by kentbower** [comment]
_**rneville** trac comment on Mar 9, 2011, 10:24:12 AM:_

(Changeset [[1285]](../commit/5407bcb2e2e405ee85b9bc371d38e3f5acb66013))
fixes #673 - System should know the rules for Extended Warranties and apply them on the fly (Smart Choice Protection is sold based on Sales Volume) [ELD]

 * If _any_ smartchoice warranty SKU is added to an order, the correct smartchoice productid will be chosen and replace any incorrect ones.
 * If the order had previously been saved and it is necessary to replace the smartchoice warranty due to price changes, product changes, etc., then the original line will be voided and a new line added with the correct smartchoice warranty
 * Smartchoice price tiers are exclusive of setup, delivery and taxes, but considered _after_ all discounts
 * Discounts are universally excluded from smartchoice products

**Comment by kentbower** [comment]
_**rneville** trac status change on Mar 9, 2011, 10:24:12 AM:_ **closed**
